components/general/Filter.vue

```diff
@@ -11,74 +15,92 @@
       />
     </div>
 
-    <div class="flex flex-wrap gap-4">
-      <Boolean name="has_image" label="Len s obrázkom" :default="true" />
-      <Boolean name="has_iip" label="Len so zoomom" />
-    </div>
-
-    <div class="flex flex-wrap gap-3">
-      <div
-        v-for="item in selected"
-        :key="`${item.key}-${item.value}`"
-        class="bg-primary text-white flex items-center py-1 px-3 gap-2 cursor-pointer rounded-3xl"
-        @click="item.toggle()"
-      >
-        <div class="text-xs">{{ item.value }}</div>
-        <Icon name="close" class="w-3" />
+    <div
+      v-show="width >= 768"
+      class="flex flex-col items-start gap-4 transition-opacity"
+      :class="{ 'opacity-0': !isReady }"
+    >
+      <div class="flex flex-wrap gap-4">
+        <Boolean name="has_image" label="Len s obrázkom" :default="true" />
+        <Boolean name="has_iip" label="Len so zoomom" />
       </div>
 
-      <div
-        v-if="selected.length"
-        class="flex py-1 px-3 gap-2 border-2 border-dark items-center cursor-pointer rounded-3xl"
-        @click="onResetAll"
-      >
-        <Icon class="w-4 h-4" name="rotate" />
-        <div class="text-xs">Zrušiť výber</div>
+      <div class="flex flex-wrap gap-3">
+        <div
+          v-if="selected.length"
+          class="flex py-1 px-3 gap-2 border-2 border-dark items-center cursor-pointer rounded-3xl"
+          @click="onResetAll"
+        >
+          <Icon class="w-4 h-4" name="rotate" />
+          <div class="text-xs">Zrušiť výber</div>
+        </div>
       </div>
     </div>
+    <FilterMobile
+      :is-opened="isMobileMenuOpened"
+      :components="components"
+      @toggleMenu="onToggleMobileMenu"
+      @resetAll="onResetAll"
+    />
   </div>
 </template>
 
 <script setup lang="ts">
+import { useWindowSize } from '@vueuse/core'
+
 import { Select, Range } from '#components'
+import SelectMobile from '~/components/controls/mobile/Select.vue'
+import RangeMobile from '~/components/controls/mobile/Range.vue'
 import Icon from '~/components/general/Icon.vue'
 import { useControls } from '~/composables/controls'
+import FilterMobile from '~/components/general/FilterMobile.vue'
 
+const isReady = ref(false)
+const isMobileMenuOpened = ref(false)
+
+const { width } = useWindowSize()
 const componentRef = ref<InstanceType<typeof Select>[]>([])
 
 const components = [
   {
     component: Select,
+    mobileComponent: SelectMobile,
     key: 'author',
     label: 'autor',
   },
   {
     component: Select,
+    mobileComponent: SelectMobile,
     key: 'work_type',
     label: 'výtvarný druh',
   },
```

Comment:
```suggestion
const components = [
  {
    component: Select,
    mobileComponent: SelectMobile,
    key: 'author',
    label: 'autor',
  },
  {
    component: Select,
    mobileComponent: SelectMobile,
    key: 'work_type',
    label: 'výtvarný druh',
  },
```

Sorry but I don't like where this is going. Please re-work into components as suggested in https://github.com/SlovakNationalGallery/sbirky-moravska-galerie/pull/1#discussion_r1418661686